### DISCUSSION TITLE
Interception android apps code

### DISCUSSION THREAD

**Discussion by kenduq** [mitmproxy/mitmproxy#5205] [Q&A]
I use mitmproxy in socks5 mode and the PCAPdroid application to track the movement of only one application in android.
I have a question: is it possible to track application traffic in mitmweb (web window) and manipulate it there?
I am trying to do a response-body interception / manipulation in mitmproxy, puropse is to enter my javascript code to display e.g. an alert window. However, when I try to intercept (execution stop) the app's response-body code, mitmproxy let all the code go through and would not show me the changes in the app.

Can you say something about this situation?

**Comment by mhils** [comment]
You need to specify an interception filter to make sure that the request (or response) is paused at the proxy before it is forwarded.
It's the same approach in both mitmproxy and mitmweb. Here's a tutorial for mitmproxy: https://docs.mitmproxy.org/stable/mitmproxytutorial-interceptrequests/
For mitmweb, it works similary, except that, instead of pressing `i`, you set the intercept filter at the top right here:
![image](https://user-images.githubusercontent.com/1019198/159976032-9bdddc78-ebc8-45de-bd98-333437de34b8.png)
